src/sentry/utils/sentryappwebhookrequests.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import
+
+import six
+import json
+
+from dateutil.parser import parse as parse_date
+from django.conf import settings
+from django.utils import timezone
+
+from sentry.utils import redis
+from sentry.models.sentryapp import VALID_EVENTS
+from sentry.models import Organization
+
+BUFFER_SIZE = 100
+
+
+class SentryAppWebhookRequestsBuffer(object):
+    """
+    Create a data structure to store basic information about Sentry App webhook requests in Redis
+    This should store the last 100 requests and last 100 errors (in different keys) for each event type, for each Sentry App
+    """
+
+    def __init__(self, sentry_app):
+        self.sentry_app = sentry_app
+
+        cluster_id = getattr(settings, "SENTRY_WEBHOOK_LOG_REDIS_CLUSTER", None)
+        if cluster_id is None:
+            self.client = redis.clusters.get("default").get_local_client(0)
+        else:
+            self.client = redis.redis_clusters.get(cluster_id)
+
+    def _convert_redis_request(self, redis_request, event):
+        """
+        Convert the request string stored in Redis to a python dict that can be returned from the errors endpoint
+        """
```

Comment:
I think it's a better practice in general that data structures output the same type that they were input — this makes it clearer for the caller, and lends towards a more generic, reusable design. This method seems like it is much more of the responsibility of the API serializer than storage.

Reply:
@tkaemming Would you recommend just just returning the result of `json.loads` the serialized string that was stored? One thing is that some info is also stored in the Redis key and I'm not sure what the best way to return that would be without somehow changing the output type

Reply:
I changed it so that in this file, I'm only getting the result of `json.loads` and then adding 1 extra field (event type, since it was stored in the Redis key), but not doing any extra formatting. The extra formatting has moved to the endpoint file